docs/book/getting-started/zenml-pro/deployments-overview.md

```diff
@@ -0,0 +1,189 @@
+
+{% hint style="info" %}
+In all of these cases the client sdk that you pip install into your development environment is the same one found here: https://pypi.org/project/zenml/
+{% endhint %}
+
+## Which Scenario is Right for You?
+
+### SaaS Deployment
+
+Choose **SaaS** if you want to get started immediately with zero infrastructure overhead.
+
+**What runs where:**
+- ZenML Server: ZenML infrastructure
+- Metadata and RBAC: ZenML infrastructure
+- Compute and Data: [Your infrastructure](https://docs.zenml.io/stacks)
+
+**Key Benefits:**
+- ⚡ Fastest setup (minutes)
+- ✅ Fully managed by ZenML
+- 🚀 Immediate production readiness
+- 💰 Minimal operational overhead
+
+**Ideal for:** Startups, teams prioritizing time-to-value and operational simplicity, organizations comfortable leveraging managed cloud services.
+
+[Learn more about SaaS deployment →](saas-deployment.md)
+
+### Hybrid SaaS Deployment
+
+Choose **Hybrid** if you need to keep sensitive metadata in your infrastructure while benefiting from centralized user management.
+
+**What runs where:**
+- ZenML Control Plane: ZenML infrastructure
+- ZenML Pro UI: ZenML infrastructure
+- ZenML Pro Server: Your infrastructure
+- RBAC: ZenML infrastructure
+- Run metadata: Your infrastructure
+- Compute and Data: [Your infrastructure](https://docs.zenml.io/stacks)
+
+**Key Benefits:**
+- 🔐 Metadata stays in your infrastructure
+- 👥 Centralized user management
+- ⚖️ Balance of control and convenience
+- 🏢 Control plane and UI fully maintained and patched by ZenML
+- ✅ Day 1 production ready
+
+**Ideal for:** Organizations with security policies requiring metadata sovereignty, teams wanting simplified identity management without full infrastructure control.
+
+[Learn more about Hybrid deployment →](hybrid-deployment.md)
+
+### Self-hosted Deployment
+
+Choose **Self-hosted** if you need complete control with no external dependencies.
+
+**What runs where:**
+- All components: [Your infrastructure](https://docs.zenml.io/stacks) (completely isolated)
+
+**Key Benefits:**
+- 🔒 Complete data sovereignty
+- 🚫 No external network dependencies
+- 🛡️ Maximum security posture
+- 📋 Full audit trail control
+
+**Ideal for:** Regulated industries (healthcare, finance, defense), government organizations, enterprises with strict data residency requirements, environments requiring offline operation.
+
+[Learn more about Self-hosted deployment →](self-hosted-deployment.md)
+
+## Common Pipeline Execution Data Flow
+
+All three deployment scenarios follow a similar pipeline execution pattern, with differences in where authentication happens and where data resides:
+
+### Standard Data Flow Steps
+
+1. **Code Execution**: You write code and run pipelines with your client SDK using Python
+
+2. **Token Acquisition**: The ZenML client fetches short-lived tokens from your ZenML workspace for:
```

Comment:
note: this only happens if you use service connectors

Reply:
so what would be a more generally correct statement